src/pages/Setting/Setting.js

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import React, { useState } from 'react';
+import SelectDropdown from './component/SelectDropdown';
+import CompleteBtn from './component/completeBtn';
+import BudgetModal from './component/BudgetModal'; // 예산 모달 컴포넌트 import
+import AllowanceModal from './component/AllowanceModal'; // 용돈 모달 컴포넌트 import
 import './Setting.scss';
 
 const Setting = () => {
-  return <div>Setting</div>;
+  const [budgetModalOpen, setBudgetModalOpen] = useState(false);
+  const [allowanceModalOpen, setAllowanceModalOpen] = useState(false);
+
+  const openBudgetModal = () => setBudgetModalOpen(true);
+  const closeBudgetModal = () => setBudgetModalOpen(false);
+
+  const openAllowanceModal = () => setAllowanceModalOpen(true);
+  const closeAllowanceModal = () => setAllowanceModalOpen(false);
```

Comment:
각 모달창을 띄우는 state를 관리할 함수는 하나로 관리 할 수 있을것 같네요
```suggestion
 const toggleBudgetModal = ()=>{
   setBudgetModalOpen((prev)=>!prev)
 }
 const toggleAllowanceModal = ()=>{
   setAllowanceModalOpen((prev)=>!prev)
 }
```

Reply:
혹은 state를 하나로 관리 하면서 지금 열려 있는 모달에 대한 state를 명시적으로 제공함으로써 modal의 isOpen 값을 조절해 줄 수도 있습니다.
```suggestion
 const [modalState, setModalState] = useState("")
 const toggleModal =(modalType)=>{
   setModalState(modalType)
 }
 ...
 // 모달창을 열기 위한 click event
 // 예산관련 모달인 경우
 onClick ={()=>toggleModal(budget)}
 
 // 예산모달 컴포넌트
 isOpen={modalState === 'budget'}
```

Reply:
2번째는 월요일날 질문하겠습니다ㅠㅠ...